docs/data/data_lifecycle.md

```diff
@@ -0,0 +1,98 @@
+# Data Lifecycle
```

Comment:
This material is more about how to move data around, it's not really about the lifecycle. Do we want to add this material knowing much if it will be changed?